src/MizarData/include/MizarData/BaselineOrComparison.h

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef MIZAR_DATA_BASELINE_OR_COMPARISON_H_
+#define MIZAR_DATA_BASELINE_OR_COMPARISON_H_
+
+#include <type_traits>
+#include <utility>
+
+namespace orbit_mizar_data {
+
+template <typename T, typename U>
+using EnableIfUConvertibleToT = std::enable_if_t<std::is_convertible_v<U, T>>;
+
+template <typename T>
+class BaselineOrComparison {
+ public:
+  ~BaselineOrComparison() = default;
+
+  const T* operator->() const { return &value_; }
+  T* operator->() { return &value_; }
+
+  const T& operator*() const { return value_; }
+  T& operator*() { return value_; }
+
+ protected:
+  template <typename U, typename = EnableIfUConvertibleToT<T, U>>
+  explicit BaselineOrComparison(U&& value) : value_(std::forward<T>(value)) {}
```

Comment:
Do you need the `enable_if`? Don't you get a compilation error anyway? I think that `enable_if` if more for SFINAE than general use.

Reply:
And in defence of this particular SFINAE use-case. It's a common way to implement a perfect-forwarding ctor.

Reply:
Ok good.